ppb/systems/renderer.py

```diff
@@ -233,14 +242,41 @@ def prepare_resource(self, game_object):
 
         surface = image.load()
         try:
-            return self._texture_cache[surface]
+            texture = self._texture_cache[surface]
         except KeyError:
             texture = SmartPointer(sdl_call(
                 SDL_CreateTextureFromSurface, self.renderer, surface,
                 _check_error=lambda rv: not rv
             ), SDL_DestroyTexture)
             self._texture_cache[surface] = texture
-            return texture
+
+        opacity = getattr(game_object, 'opacity', 255)
+        opacity_mode = getattr(game_object, 'opacity_mode', 'blend')
+        color = getattr(game_object, 'color', (255, 255, 255))
+
+        sdl_call(
+            SDL_SetTextureAlphaMod, texture.inner, opacity,
+            _check_error=lambda rv: rv < 0
+        )
+        if opacity_mode == 'add':
```

Comment:
Direct implementation is less of a big deal, but I'd definitely avoid magic strings if we can. Either make these class attributes, exported constants, or Flags. (Or enums, but want to check with some educators before we go too far down that route.)